### PR TITLE
adding search Timer

### PR DESCRIPTION
**Pull request by natashaarmbrust**
- adds timer to delay searching

### CODE REVIEW

File: Podcast/SearchViewController.swift [hunk 1]
```diff
@@ -240,21 +241,36 @@ class SearchViewController: ViewController, UISearchControllerDelegate, UITableV
...
-            data.fetchData(query: searchText, newSearch: true)
+        if self.lastSearchText == searchText && pastSearchesTableView.isHidden { return }
+        self.lastSearchText = searchText
+        self.tabUnderlineView.isHidden = false
+        self.searchResultsTableView.isHidden = false
+        self.pastSearchesTableView.isHidden = true
+        self.didDismissItunesHeaderForQuery = false
+        self.addSearchITunesHeader()
+        self.mainScrollView = self.searchResultsTableView
+        self.searchResultsTableView.startLoadingAnimation()
```
Comment (mindylou) [inline Podcast/SearchViewController.swift]: why do these all use `self`?

File: Podcast/SearchViewController.swift [hunk 2]
```diff
@@ -240,21 +241,36 @@ class SearchViewController: ViewController, UISearchControllerDelegate, UITableV
...
+
+        self.searchResultsTableView.reloadData()
+        self.searchResultsTableView.loadingAnimation.bringSubview(toFront: self.searchResultsTableView)
+        self.updateTableViewInsetsForAccessoryView()
+
+    }
+
+    @objc func searchAfterDelay() {
+        for data in self.tableViewData {
+            data.fetchData(query: lastSearchText, newSearch: true)
```
Comment (mindylou) [inline Podcast/SearchViewController.swift]: the "no results found" screen shows up for a little bit after the loading indicator spins for a bit, and then shows results (only for series and episodes):
image

**Review by mindylou — COMMENTED** [review]
looks good! just one small bug that I found with searching/null states

File: Podcast/SearchViewController.swift (hunk 1 above)
Comment (natashaarmbrust) [reply Podcast/SearchViewController.swift]: thanks for catching that! I had it in a completion block before

**Review by doctor3w — APPROVED** [review]
LGTM

**Pull request merged**